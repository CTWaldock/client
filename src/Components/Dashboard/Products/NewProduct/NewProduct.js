import React from 'react'
import {Body, Button} from './NewProduct.styles'
import Form from '../../../../UI/Form'
import FormField from '../../../../UI/FormField'
import TextArea from '../../../../UI/TextArea'



const NewProduct = (props) => {
    console.log(props)
    return(
        <Body>
            <div>
                <Form onSubmit={(e) => props.onSubmit(e)}>
                    <FormField  label="Product Name" name="name" type="text" onChange="" />
                    <FormField  label="Product Id" name="productId" type="text" onChange="" />
                    <FormField  step=".01" label="Product Price" name="price" type="number" onChange="" />
                    <FormField  label="Product QTY" name="stockQty" type="number" onChange="" />
                    <TextArea label="Product Description" name="description" type="text" onChange="" />

                    <Button>Create new product</Button>
                </Form>
            </div>
        </Body>
    )
}


export default NewProduct
