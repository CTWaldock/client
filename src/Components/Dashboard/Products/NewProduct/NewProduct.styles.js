import styled from 'styled-components'
import { styleVariables } from '../../../../Assets/Styles';
const { colors } = styleVariables

const Body = styled.div`
    background: black;
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
`


export {
    Body
}
