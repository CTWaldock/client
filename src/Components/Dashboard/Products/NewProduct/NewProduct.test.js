import React from 'react'
import NewProduct from './NewProduct'
import renderer from 'react-test-renderer'



describe('NewProduct', () => {
    it('Should render the way we expect', () => {


        const tree = renderer.create(
                <NewProduct onSubmit={() => {}} />
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
