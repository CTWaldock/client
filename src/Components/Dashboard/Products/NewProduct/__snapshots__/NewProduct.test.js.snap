// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProduct Should render the way we expect 1`] = `
<div
  className="sc-bZQynM isBous"
>
  <div>
    <form
      className="sc-dnqmqq kOFvKe"
      id={undefined}
      onSubmit={[Function]}
    >
      <h3
        className="sc-iwsKbI cWsnOw"
      />
      <label
        className="sc-gZMcBi fSOcE"
      >
        Product Name
        <span />
      </label>
      <input
        autoComplete="nope"
        autoFocus={null}
        className="sc-gqjmRU ifBdEQ"
        name="name"
        onChange={null}
        placeholder={undefined}
        step={null}
        type="text"
      />
      <label
        className="sc-gZMcBi fSOcE"
      >
        Product Id
        <span />
      </label>
      <input
        autoComplete="nope"
        autoFocus={null}
        className="sc-gqjmRU ifBdEQ"
        name="productId"
        onChange={null}
        placeholder={undefined}
        step={null}
        type="text"
      />
      <label
        className="sc-gZMcBi fSOcE"
      >
        Product Price
        <span />
      </label>
      <input
        autoComplete="nope"
        autoFocus={null}
        className="sc-gqjmRU ifBdEQ"
        name="price"
        onChange={null}
        placeholder={undefined}
        step=".01"
        type="number"
      />
      <label
        className="sc-gZMcBi fSOcE"
      >
        Product QTY
        <span />
      </label>
      <input
        autoComplete="nope"
        autoFocus={null}
        className="sc-gqjmRU ifBdEQ"
        name="stockQty"
        onChange={null}
        placeholder={undefined}
        step={null}
        type="number"
      />
      <label
        className="sc-VigVT dkAinW"
      >
        Product Description
        <span />
      </label>
      <textarea
        autoComplete="nope"
        autoFocus={null}
        className="sc-jTzLTM fKHxiT"
        name="description"
        onChange={null}
        placeholder={undefined}
        type="text"
      />
      <button
        className="sc-gzVnrw uzLKJ"
      >
        Create new product
      </button>
      <div />
    </form>
  </div>
</div>
`;
